src/de/mod.rs

```diff
@@ -152,7 +149,13 @@ fn read_cstring<R: Read + ?Sized>(reader: &mut R) -> Result<String> {
         v.push(c);
     }
 
-    Ok(String::from_utf8(v)?)
+    to_string(v)
+}
+
+fn to_string(v: Vec<u8>) -> Result<String> {
+    let _ = simdutf8::basic::from_utf8(&v)?;
+    // Safety: `v` is a valid UTF-8 string.
+    unsafe { Ok(String::from_utf8_unchecked(v)) }
```

Comment:
Any reason the result of the first line of this function cannot be converted to a `String` and returned directly?

```
fn to_string(v: Vec<u8>) -> Result<String> {
    let str = simdutf8::basic::from_utf8(&v)?;
    Ok(str.to_string())
}
```
I'd like to avoid unsafe code if we do not need it here.

Reply:
`str.to_string() ` will make a copy of original bytes. It may be even slower then the std's `String::from_utf8`.

Also, you can see the issue [here](https://github.com/rusticstuff/simdutf8/issues/73).